vernac/command.ml

```diff
@@ -130,8 +130,8 @@ let interp_definition pl bl poly red_option c ctypopt =
         in
 	if not (try List.for_all chk imps2 with Not_found -> false)
 	then warn_implicits_in_term ();
-        let vars = Univ.LSet.union (Univops.universes_of_constr body) 
-          (Univops.universes_of_constr typ) in
+        let vars = Univ.LSet.union (EConstr.universes_of_constr !evdref (EConstr.of_constr body))
+          (EConstr.universes_of_constr !evdref (EConstr.of_constr typ)) in
```

Comment:
This union is getting pretty big, how about let-lifting the universes_of_constr calls out?

Reply:
Yep, good catch